ui-tests/tests/jupyterlab_jhub_apps.spec.ts

```diff
@@ -137,3 +199,29 @@ test.describe('should register custom commands', () => {
     expect(deployAppMainMenu.isEnabled).toBe(true);
   });
 });
+
+test('should restore correct state', async ({ page }) => {
+  await page.notebook.createNew('Untitled.ipynb');
+  await page.getByLabel('Untitled.ipynb').getByText('Deploy App').click();
+  expect(
+    page.getByRole('tab', { name: 'Deploy App', selected: true })
+  ).toBeVisible();
+
+  await page.notebook.createNew('Untitled1.ipynb');
+  await page.getByLabel('Untitled1.ipynb').getByText('Deploy App').click();
+  expect(
+    page.getByRole('tab', { name: 'Deploy App', selected: true })
+  ).toBeVisible();
+
+  // opens the same form again - this should not be restored
+  // below as duplicates should not be restored
+  await page.getByRole('tab', { name: 'Untitled1.ipynb' }).click();
+  await page.getByLabel('Untitled1.ipynb').getByText('Deploy App').click();
+  expect(
+    page.getByRole('tab', { name: 'Deploy App', selected: true })
+  ).toBeVisible();
+
+  await page.reload();
```

Comment:
This is currently causing me a problem. When I run the tests in debug mode, and add a `page.waitForTimeout(1000)`, the restore functionality works fine. But if I remove the timeout, it no longer restores correctly.

Ideally I would wait for some signal that indicates that  restore has successfully stored the state, before calling `page.reload()`. But I'm not sure what signal this is...